crates/core_arch/src/aarch64/neon/mod.rs

```diff
@@ -8,25 +8,24 @@ pub use self::generated::*;
 // FIXME: replace neon with asimd
 
 use crate::{
-    core_arch::{arm::*, simd_llvm::*},
+    core_arch::{arm::*, simd::*, simd_llvm::*},
     mem::{transmute, zeroed},
 };
 #[cfg(test)]
 use stdarch_test::assert_instr;
 
+pub type p64 = u64;
+pub type p128 = u128;
+
 types! {
-    /// ARM-specific 64-bit wide vector of one packed `f64`.
-    pub struct float64x1_t(f64); // FIXME: check this!
-    /// ARM-specific 128-bit wide vector of two packed `f64`.
-    pub struct float64x2_t(f64, f64);
     /// ARM-specific 64-bit wide vector of one packed `p64`.
-    pub struct poly64x1_t(i64); // FIXME: check this!
+    pub struct poly64x1_t(p64); // FIXME: check this!
     /// ARM-specific 64-bit wide vector of one packed `p64`.
-    pub struct poly64_t(i64); // FIXME: check this!
+    pub struct poly64_t(p64); // FIXME: check this!
     /// ARM-specific 64-bit wide vector of two packed `p64`.
-    pub struct poly64x2_t(i64, i64); // FIXME: check this!
-    /// ARM-specific 128-bit wide vector of one packed `p64`.
-    pub struct poly128_t(i128); // FIXME: check this!
+    pub struct poly64x2_t(p64, p64); // FIXME: check this!
+    /// ARM-specific 128-bit wide vector of one packed `p128`.
+    pub struct poly128_t(p128); // FIXME: check this!
```

Comment:
These types should also be available on ARM: they are used by the crypto extension which is supported on ARMv8 in both 32-bit and 64-bit modes.